### PR TITLE
Create input field component

### LINKED ISSUE #2705: (SP: 5) Create InputField component aligned with design system
#### Description:
Implement a reusable `InputField` component that meets the design system standards. This component will be used throughout the application to maintain consistency in input field styles, behavior, and accessibility.

[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=21376-87013&t=l5PI8JRXpNFKQBr0-4)

#### Requirements:
- **Styles:**
  - Follow design system variables for colors, typography, padding, and other styles.
  - Support different variants (large, small, outlined).
  - Ensure hover, focus, active, and disabled states are defined according to the design system.
  - Provide a label for inputField.
  
- **Props:**
  - `variant`: Defines type (large, small, outlined).
  - `label`: Provide a label for the inputField.
  - `disabled`: Disables the inputField when true.
  - `value`: Provides the value to the component.
  - `placeholder`:  Supports placeholder for component.
  - `helperText`: Supports helperText in giving user hints or warnings.
  - `error`: Supports error text and error icon in giving user hints or warnings when true
  - `onChange` : A callback function to handle input changes.
  - `search` : adds a search icon when true
  - `sx`: The sx prop will accept a style object that can be merged with the existing styles.

Don't forget to cover component with tests and add to the storybook

### PR DESCRIPTION
**Pull request by PavloButynets**
Implemented a reusable InputField component that meets the standards of the design system. This component will be used throughout the application to maintain consistency in input field styles, behavior, and accessibility. Added it to the storybook, wrote tests for it
https://youtu.be/rINxNyt-b8E

### CODE REVIEW

File: src/design-system/components/input-field/InputField.scss [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+    input {
+      color: get-var('red-500');
+    }
+    .#{$prefix}input-label {
+      color: get-var('red-500');
+    }
+  }
+
+  input::placeholder {
```
Comment (amoutens) [inline src/design-system/components/input-field/InputField.scss]: you can add this opacity setting to 133 line

**Review by amoutens — CHANGES_REQUESTED** [review]

**Review by amoutens — APPROVED** [review]

**Comment by Mav-Ivan** [comment]
teammate that add comment should resolve it 
image

**Review by Mav-Ivan — APPROVED** [review]

**Review by SofiiaYevush — APPROVED** [review]

File: src/design-system/components/input-field/InputField.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const clearInput = () => {
+    onChange({ target: { value: '' } } as React.ChangeEvent<HTMLInputElement>)
+  }
+
+  return (
+    <Box sx={sx as SxProps}>
+      <div
+        className={`s2s-input-container s2s-input-container_${variant} ${
+          disabled ? 's2s-input-container_disabled' : ''
+        } ${error ? 's2s-input-container_error' : ''}`}
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/input-field/InputField.tsx]: Use `cn` function here

File: src/design-system/components/input-field/InputField.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        <label className='s2s-input-label'>{label}</label>
+        {error ? (
+          <ErrorOutlineIcon className='s2s-error-icon' />
+        ) : (
+          <ClearIcon className='s2s-clear-icon' onClick={clearInput} />
+        )}
+      </div>
+      {helperText && (
+        <div className={`s2s-helper-text-container`}>
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/input-field/InputField.tsx]:
```suggestion
        <div className='s2s-helper-text-container'>
```

File: src/design-system/components/input-field/InputField.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+          <span className={error ? 's2s-helper-text-error' : 's2s-helper-text'}>
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/input-field/InputField.tsx]: Use `cn` function here

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: src/design-system/components/input-field/InputField.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    onChange({ target: { value: '' } } as React.ChangeEvent<HTMLInputElement>)
+  }
+
+  return (
+    <Box sx={sx as SxProps}>
+      <div
+        className={cn('s2s-input-container', `s2s-input-container_${variant}`, {
+          's2s-input-container_disabled': disabled,
+          's2s-input-container_error': error
+        })}
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/input-field/InputField.tsx:47]: I think using `cn` this way (without object) will be more readable and simple, but it's up to you
```suggestion
        className={cn('s2s-input-container', 
        `s2s-input-container_${variant}`,
        disabled && 's2s-input-container_disabled',
        error && 's2s-input-container_error'
        )}
```

File: src/design-system/components/input-field/InputField.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        ) : (
+          <ClearIcon className='s2s-clear-icon' onClick={clearInput} />
+        )}
+      </div>
+      {helperText && (
+        <div className='s2s-helper-text-container'>
+          <span
+            className={cn('s2s-helper-text', {
+              's2s-helper-text-error': error
+            })}
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/input-field/InputField.tsx:71]: Same here
I think that you should use object in `cn` function when your classname has the same name as condition. For example - classname = "error" and condition = error, so technically object is better for this example. But right now you have different names for classname and condition, and there is no reason to use object here.
But as I sad before, it's up to you)

**Review by ShadowOfTheSpace — APPROVED** [review]

**Pull request merged**